### PR TITLE
Add ml-api-common to capi

### LINKED ISSUE #75: Split Error Code
The Error Code in Tizen API is duplicated with NNStreamer.
Therefore need to make common package for the common header to remove the conflict and include that package when nntrainer build.

### PR DESCRIPTION
**Pull request by zhoonit**
**Changes proposed in this PR:**
- Add capi-ml-common-devel to spec file
- Add `ml-api-common.h` for dummy
- Change error code accordingly

Resolves #75

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

### CODE REVIEW

File: api/capi/include/nntrainer.h [hunk 1]
```diff
@@ -27,7 +27,11 @@
 #include <stdbool.h>
 #include <stddef.h>
 
-#include <tizen_error.h>
+#if defined(__TIZEN__)
+#include <nnstreamer/ml-api-common.h>
+#else
+#include "ml-api-common.h"
+#endif
```
Comment (myungjoo) [inline api/capi/include/nntrainer.h:34]: Be careful, Tizen API auditors won't like this. 

Recommendation:
Use ```#include <ml-api-common.h>``` (nnstreamer.pc has ```Cflags: -I${includedir}/nnstreamer```)
Remove ```#if / #endif```
Update meson.build to set api/capi/include as includedir (allow to use ```<ml-api-common.h>``` for this directory) if it is not Tizen and try to disable it (move away or rename it) if it is Tizen.

Comment (zhoonit) [reply api/capi/include/nntrainer.h]: Thanks, it is fixed. Please take another look :)

File: api/capi/include/platform/ml-api-common.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  ML_ERROR_NONE                 = TIZEN_ERROR_NONE, /**< Success! */
+  ML_ERROR_INVALID_PARAMETER    = TIZEN_ERROR_INVALID_PARAMETER, /**< Invalid parameter */
+  // ML_ERROR_STREAMS_PIPE         = TIZEN_ERROR_STREAMS_PIPE, /**< Cannot create or access the pipeline. not used in nntrainer */ 
+  // ML_ERROR_TRY_AGAIN            = TIZEN_ERROR_TRY_AGAIN, /**< The pipeline is not ready, yet (not negotiated, yet) not used in nntrainer */
+  ML_ERROR_UNKNOWN              = TIZEN_ERROR_UNKNOWN,  /**< Unknown error */
+  ML_ERROR_TIMED_OUT            = TIZEN_ERROR_TIMED_OUT,  /**< Time out */
+  ML_ERROR_NOT_SUPPORTED        = TIZEN_ERROR_NOT_SUPPORTED, /**< The feature is not supported */
+  ML_ERROR_PERMISSION_DENIED    = TIZEN_ERROR_PERMISSION_DENIED, /**< Permission denied */
+  ML_ERROR_OUT_OF_MEMORY        = TIZEN_ERROR_OUT_OF_MEMORY, /**< Out of memory (Since 6.0) */
+} ml_error_e;
```
Comment (myungjoo) [inline api/capi/include/platform/ml-api-common.h:41]: 1. Add defensive mechanism:
E.g.,
```
#ifdef __TIZEN__
#error TIZEN build should not use this
#endif
```

2. Define error code directly
e.g.,
```
#define ML_ERROR_OUT_OF_MEMORY    (ENOMEM)
```

File: api/capi/meson.build [hunk 3]
```diff
@@ -1,7 +1,10 @@
 capi_inc = []
 capi_inc += include_directories('include')
 
-if not get_option('enable-tizen')
+# pc file is not present for 'ml-api-common' yet
+if cxx.has_header('nnstreamer/ml-api-common.h', required: false)
+  capi_inc += include_directories ('/usr/include/nnstreamer')
+else
```
Comment (myungjoo) [inline api/capi/meson.build:7]: this file LGTM

File: api/capi/include/platform/ml-api-common.h (hunk 2 above)
Comment (myungjoo) [reply api/capi/include/platform/ml-api-common.h:41]: You don't need to have a separated file tizen_error.h as long as you have this file for non-Tizen platforms. Merge them (by the comment above)

Comment (zhoonit) [reply api/capi/include/platform/ml-api-common.h:57]: Thank you for the review. It is fixed :)

File: api/capi/include/platform/ml-api-common.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ *    other modules of Tizen-ML APIs use common data structures.
+ */
+#ifndef __ML_API_COMMON_H__
+#define __ML_API_COMMON_H__
+
+#ifdef __TIZEN__
+
+#error TIZEN build should not use this file.
+
+#else
```
Comment (myungjoo) [inline api/capi/include/platform/ml-api-common.h:26]: #endif. Try to lower the if/then/else depth.

File: api/capi/include/platform/ml-api-common.h (hunk 2 above)
Comment (myungjoo) [reply api/capi/include/platform/ml-api-common.h:57]: "2." is not reflected.

Comment (zhoonit) [reply api/capi/include/platform/ml-api-common.h]: Uhh I misunderstood "2." Fixed!

File: api/capi/include/platform/ml-api-common.h (hunk 4 above)
Comment (zhoonit) [reply api/capi/include/platform/ml-api-common.h]: Much better. Thank you!

File: api/capi/meson.build [hunk 5]
```diff
@@ -11,6 +14,7 @@ capi_src += join_paths(meson.current_source_dir(), 'src','nntrainer.cpp')
 capi_headers = []
 capi_headers += join_paths(meson.current_source_dir(), 'include', 'nntrainer.h')
 
+
```
Comment (wooksong) [inline api/capi/meson.build:17]: Unnecessary blank line?

File: api/capi/include/platform/ml-api-common.h [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+#endif /** __TIZEN__ **/
+
+#ifdef __cplusplus
+extern "C" {
+#endif /* __cplusplus */
+
+#include <errno.h>
+#define ML_ERROR_NONE (0)
+#define ML_ERROR_INVALID_PARAMETER (-EINVAL)
+#define ML_ERROR_UNKNOWN (-1073741824LL)
```
Comment (wooksong) [inline api/capi/include/platform/ml-api-common.h:35]: How about using 0x notation here for better readability?

Comment (zhoonit) [reply api/capi/include/platform/ml-api-common.h]: Good idea. Thanks :)

File: api/capi/meson.build (hunk 5 above)
Comment (zhoonit) [reply api/capi/meson.build]: Fixed. Thank you 👍

**Pull request merged**